web/thesauruses/javascript/ECMAScript 2023/strings.json

```diff
@@ -0,0 +1,352 @@
+        "name": "Convert string to title case",
+        "code": "str.replace(/\\w+/g, (w) => w[0].toUpperCase() + w.slice(1).toLowerCase())"
+      },
+      "capitalize_string": {
+        "name": "Capitalize first letter of a string",
+        "code": [
+            "str[0].toUpperCase() + str.slice(1)",
+            "str.replace(/^\\w/, (w) => w.toUpperCase())"
+        ]
+      },
+      "remove_whitespace": {
+        "name": "Remove all whitespaces from string",
+        "code": "str.replace(/\\s/g, \"\")"
+      },
+      "replace_substring": {
+        "name": "Replace a substring with another string",
+        "code": [
+            "str.replace(substring, another)",
+            "str.split(substring).join(another)"
+        ]
+      },
+      "replace_all_substring": {
+        "name": "Replace all substring matches with another string",
+        "code": [
+            "str.replace(/substring/g, another)",
+            "str.split(substring).join(another)"
+        ]
+      },
+      "split_at_index": {
+        "name": "Split string into a list of strings at a given index",
+        "code": "str.split(separator, limit)",
+        "comment": ""
+      },
+      "split_at_newlines": {
+        "name": "Split string into a list of strings at every new line character",
+        "code": "str.split(\"\\n\")"
+      },
+      "split_at_substring": {
+        "name": "Split string by locating all substrings",
+        "code": [
+            "str.split(substring)",
+            "str.match(/substring/g)"
+        ]
+      },
+      "merge_lists_into_string": {
+        "name": "Merge a list of strings into one string",
+        "code": [
+            "list.join(\" \")",
+            "list.reduce((a, b) => a + \" \" + b)"
+        ]
+      },
+      "encode_html_entities": {
+        "name": "Encode HTML entities in a string (ex: \u2122 to &trade;)",
+        "code": "str.replace(/[\\u00A0-\\u9999<>\\&]/gim, (i) => `&#${i.charCodeAt(0)};`)"
+      },
+      "decode_html_entities": {
+        "name": "Decode HTML entitles into characters",
+        "code": "(new DOMParser).parseFromString(str, \"text/html\").documentElement.textContent"
+      },
+      "encode_url_percent": {
+        "name": "Encode URL percent encoding into string (ex: ' ' to %20)",
+        "code": [
+          "encodeURI(str)",
+          "encodeURIComponent(str)"
+        ]
+      },
+      "decode_url_percent": {
+        "name": "Decode URL percent encoding",
+        "code": [
+          "decodeURI(str)",
+          "decodeURIComponent(str)"
+        ]
+      },
+      "encode_to_base64": {
+        "name": "Encode string into Base64 format",
+        "code": "btoa(str)"
+      },
+      "decode_from_base64": {
+        "name": "Decode string from Base64 format",
+        "code": "atob(str)"
+      },
+      "format_string_function": {
+        "name": "Function to format a string",
+        "code": [
+          "'' + str",
+          "String(str)"
+        ]
+      },
```

Comment:
This is for like functions to format strings in various ways. You are just coercing things into strings, which is not the same. It looks like the various `*.format()` functions you ahve below are what you use, but since they all vary and it's not one function to do them all, I think this is `not-implemented`, right?

Reply:
Yes, I'll add `not-implemented`!